### PR TITLE
Bibliography and other adjustments

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [x] The relevant author information has been added to `AUTHORS.rst` and `.zenodo.json`

### What kind of change does this PR introduce?

* Adds `pybtex` to development dependencies (used in docs/conf.py)
* Reverts the conditional bibliography changes for all builds of docs
* Removes the latex-generated bibliography automatically generated in `latexpdf` mode

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@aulemahal feel free to add your docs changes in here!

File: .readthedocs.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ build:
       - pip install llvmlite~=0.38 --ignore-installed
     pre_build:
       - sphinx-apidoc -o docs/ --private --module-first xclim xclim/testing/tests
-      - env SKIPNOTEBOOKS=1 sphinx-build -b linkcheck docs/ _build/linkcheck
+      - env SKIPNOTEBOOKS=1 sphinx-build -b linkcheck docs/ _build/linkcheck || true
```
Comment (aulemahal) [inline .readthedocs.yml:19]: What does this do? Do I understand it ensures the bash command always returns "successfully" ?

**Review by aulemahal — APPROVED** [review]
Nice!

File: .readthedocs.yml (hunk 1 above)
Comment (Zeitsperre) [reply .readthedocs.yml:19]: Essentially. The example necessitating it here is that dask.org was down today (it recently went back up). We know this is the proper link, so we don't want it to crash. The command here is now more of a verbose warning.

**Pull request merged**